include/utils.h

```diff
@@ -0,0 +1,12 @@
+#ifndef UTILS_H
```

Comment:
Sure. The idea was to separate out the helper functions for now to reduce clutter